srml/support/procedural/src/storage/genesis_config.rs

```diff
@@ -0,0 +1,349 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use srml_support_procedural_tools::syn_ext as ext;
+use proc_macro2::TokenStream as TokenStream2;
+
+use syn::{WhereClause, spanned::Spanned};
+use quote::{quote, quote_spanned};
+
+use super::instance_trait::DEFAULT_INSTANTIABLE_TRAIT_NAME;
+
+const DEFAULT_INSTANCE_NAME: &str = "__GeneratedInstance";
+
+pub fn impl_genesis_config(scrate: &TokenStream2, def: &super::DeclStorageDefExt) -> TokenStream2 {
```

Comment:
Refactor and introducing a 320 lines function, is no "refactoring" ;) No one is able to read/understand this.

Reply:
Yes actually this function hasn't been really touched, just moved. but ok I can do it here as well.

Reply:
Ok improved